torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -1423,14 +1423,16 @@ std::vector<c10::Device> TensorPipeAgent::getDevicesForRemote(
   if (iter == deviceMaps.end()) {
     for (const auto& t : message.tensors()) {
       TORCH_CHECK(
-          t.device().is_cpu(),
+          t.device().is_cpu() || t.device().is_meta(),
           errStr,
           ", but found tensor on device: ",
           t.device());
     }
-    return {};
+    return getDevicesForTensors(message.tensors(), {{c10::kCPU, c10::kCPU}, {c10::kMeta, c10::kMeta}}, errStr);
   } else {
-    return getDevicesForTensors(message.tensors(), iter->second, errStr);
+    DeviceMap mergedMap = iter->second;
+    mergedMap.insert({{c10::kCPU, c10::kCPU}, {c10::kMeta, c10::kMeta}});
```

Comment:
does it make sense to insert meta -> meta by default into the device map instead of patching it here?

Reply:
reverted